### DISCUSSION TITLE
PDP gallery, swatch selection error

### DISCUSSION THREAD

**Discussion by rutulritz** [breezefront/community#108] [1. Help]
Hi,

I am getting below error when trying to select color option swatch click and the main image is not updating in the gallery even thumbnails are not loading as per swatch selection.
you can check here : https://eeupgrade.katespade.co.id/morgan-flap-chain-wallet-0k4007256-k8925.html
![image](https://user-images.githubusercontent.com/9527439/232390557-019a53d4-8924-4dcf-8c1b-8ff9b848f1af.png)

This error is showing on page load also. Can you please help me on this?

Thanks

**Comment by vovayatsyuk** [comment]
Some of the products don't have the required image data. These product ids are 443972, 443975, 443981.

<img src="https://user-images.githubusercontent.com/306080/232433229-38405957-4165-46fa-83cc-0001f3277d41.png" width="298">

I don't know why. Maybe product data wasn't properly imported into Magento.

**Reply by rutulritz** [reply]
Ok got it. Thanks.

One more issue here. https://id.coach.com/willow-bucket-bag-in-colorblock-d01lthshl-c3766.html

If I am selecting particular color then it should show only related color product thumbnail instead of showing all images. 
Is it the default functionality of breeze gallery or any issue in my integration of breeze gallery on PDP ? 

Can you please guide me?

Thanks

**Reply by vovayatsyuk** [reply]
If I remember correctly this is the Magento feature, and it's configurable via this option: https://github.com/breezefront/theme-frontend-breeze-blank/blob/master/etc/view.xml#L268-L270

Try to change `gallery_switch_strategy` to `replace` in your view.xml file.

**Reply by rutulritz** [reply]
It is working @vovayatsyuk . 
Thank you.

**Comment by rutulritz** [comment]
Hi @vovayatsyuk ,

For above product image missing issue, How can I fix this if there are missing images in srcset?

Can we fix this error? Please guide me

Thanks

**Reply by vovayatsyuk** [reply]
Try to open these products in the backend and set the full and thumbnail images for them.

**Reply by rutulritz** [reply]
Hi @vovayatsyuk

Here I checked for this PDP  : https://katespade.co.id/knott-small-crossbody-0k5007076-k6554.html

I have checked the base and thumbnail images from backend it is already set as per below screenshots.

![ks-2](https://user-images.githubusercontent.com/9527439/233310502-a5bb4fd6-de67-4683-a05e-65033eedd0b3.png)
![ks-1](https://user-images.githubusercontent.com/9527439/233310780-f5119c85-dc3c-4f6b-a3af-5fce2b1d9247.png)

So why still showing error. Can you please look into this once and help me?

Thanks

**Reply by vovayatsyuk** [reply]
Thanks for the screenshot! I was able to find a bug in the breeze responsive images implementation. Fixed in https://github.com/breezefront/module-breeze/commit/7a7a35230d5dd4dc781ca017cf4ee738709c1b47

**Reply by rutulritz** [reply]
Thank you @vovayatsyuk , It is working now.

**Comment by rutulritz** [comment]
Hi @vovayatsyuk ,

After above changes swatch selection is working fine on PDP. But, on PLP the images is showing broken (it should be in full size placeholder same as others) for some products. Can you please check below link : 
https://eeupgrade.katespade.co.id/handbags/category/view-all.html
Admin Images : 
![image](https://user-images.githubusercontent.com/9527439/234469787-1c39dc72-dd5c-4178-aa3e-d28866cb1fcc.png)


![image](https://user-images.githubusercontent.com/9527439/234467494-ab9ec98d-2017-41ce-9faa-e7159995bc8f.png)


Please help me on this.

Thanks

**Reply by vovayatsyuk** [reply]
Maybe your placeholder is too small to show it in the same size? Try to use a bigger placeholder image.

**Reply by rutulritz** [reply]
Ok got it thanks. 

One more thing in some of the product thu thumbnails are rendering from configurabe product instead of loading it from simple product images. 
Can you please check below urls for reference : 
https://katespade.co.id/0k5014078-pxr00466.html
https://id.coach.com/dakota-bucket-bag-d01mixshb-ch739.html


Here it should show only brown color images from simple product's mages, but showing all the images from the configurable item: 
below are the screenshot for the config and simple product images : 

Simple images
![simple-product-images](https://user-images.githubusercontent.com/9527439/235124868-9eb61f3c-3279-4f1b-aeda-fe64d1a4805f.png)

Configurable images
![config-product-images](https://user-images.githubusercontent.com/9527439/235124832-146cfa19-7851-4345-85ba-569532870230.png)

Can you help me to understand this or need any changes to do for fix this ? 

Thanks

**Reply by rutulritz** [reply]
@vovayatsyuk , I am waiting for your reply. 

Thanks

**Reply by vovayatsyuk** [reply]
Looks like it was a Magento bug, that was fixed in 2.4.5. Added the same fix into our code: https://github.com/breezefront/module-breeze/commit/4f1f7abe216b28a3f2cfb5bc78c8f69b65193820

Can you please try if it works fine?
